src/cdk-experimental/listbox/listbox.ts

```diff
@@ -0,0 +1,157 @@
+    '(keydown)': 'pattern.onKeydown($event)',
+    '(pointerdown)': 'pattern.onPointerdown($event)',
+  },
+})
+export class CdkListbox {
+  /** The directionality (LTR / RTL) context for the application (or a subtree of it). */
+  private readonly _directionality = inject(Directionality);
+
+  /** The CdkOptions nested inside of the CdkListbox. */
+  private readonly _cdkOptions = contentChildren(CdkOption, {descendants: true});
+
+  /** A signal wrapper for directionality. */
+  protected textDirection = toSignal(this._directionality.change, {
+    initialValue: this._directionality.value,
+  });
+
+  /** The Option UIPatterns of the child CdkOptions. */
+  protected items = computed(() => this._cdkOptions().map(option => option.pattern));
+
+  /** Whether the list is vertically or horizontally oriented. */
+  orientation = input<'vertical' | 'horizontal'>('vertical');
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable = input(false, {transform: booleanAttribute});
+
+  /** Whether focus should wrap when navigating. */
+  wrap = input(true, {transform: booleanAttribute});
+
+  /** Whether disabled items in the list should be skipped when navigating. */
+  skipDisabled = input(true, {transform: booleanAttribute});
+
+  /** The focus strategy used by the list. */
+  focusMode = input<'roving' | 'activedescendant'>('roving');
+
+  /** The selection strategy used by the list. */
+  selectionMode = input<'follow' | 'explicit'>('follow');
+
+  /** The amount of time before the typeahead search is reset. */
+  typeaheadDelay = input<number>(0.5); // Picked arbitrarily.
+
+  /** Whether the listbox is disabled. */
+  disabled = input(false, {transform: booleanAttribute});
+
+  // TODO(wagnermaciel): Figure out how we want to expose control over the current listbox value.
+  /** The ids of the current selected items. */
+  selectedIds = model<string[]>([]);
+
+  /** The current index that has been navigated to. */
+  activeIndex = model<number>(0);
+
+  /** The Listbox UIPattern. */
+  pattern: ListboxPattern = new ListboxPattern({
+    ...this,
+    items: this.items,
+    textDirection: this.textDirection,
+  });
+}
+
+/** A selectable option in a CdkListbox. */
+@Directive({
+  selector: '[cdkOption]',
+  exportAs: 'cdkOption',
+  host: {
+    'role': 'option',
+    'class': 'cdk-option',
+    '[attr.tabindex]': 'pattern.tabindex()',
+    '[attr.aria-selected]': 'pattern.selected()',
+    '[attr.aria-disabled]': 'pattern.disabled()',
+  },
+})
+export class CdkOption {
+  /** A reference to the option element. */
+  private readonly _elementRef = inject(ElementRef);
+
+  /** The parent CdkListbox. */
+  private readonly _cdkListbox = inject(CdkListbox);
+
+  /** A unique identifier for the option. */
+  private readonly _generatedId = inject(_IdGenerator).getId('cdk-option-');
+
+  /** A unique identifier for the option. */
+  protected id = computed(() => this._generatedId);
```

Comment:
This can be a TODO / cleanup for a next PR, but these generally shouldn't be `computed` for things that won't change.

I think the reason that you're doing this is because the e.g. `OptionInputs` expects a signal. However, you don't really want those to be a `Signal<T>`, but instead a `() => T` (which would also make the types work for Wiz) . You could optionally introduce a convenience type for this that's something like

```ts
export type Reactive<T> = () => T;
```

We probably want to do this before build too many more `___Input` types